package com.example.network.datas.orders;

import com.example.network.domains.apis.MyAsyncTask;
import com.example.network.domains.callbacks.MyResponseCallback;
import com.example.network.domains.common.Settings;

import org.jsoup.Jsoup;
import org.jsoup.Connection;

import java.io.IOException;

public class OrderGet extends MyAsyncTask {


    Integer id;
    String token;

    public OrderGet(Integer id, String token, MyResponseCallback callback) {

        super(callback);
        this.id = id;
        this.token = token;

    }

    @Override
    protected String doInBackground(Void... voids) {

        try {

            Connection.Response response = Jsoup.connect(Settings.url("/api/order/get/" + id))
                    .ignoreContentType(true)
                    .ignoreHttpErrors(true)
                    .method(Connection.Method.GET)
                    .header("Content-type", "application/json")
                    .header("token", token)
                    .execute();

            return response.statusCode() == 200 ? response.body() : "Error: " + response.body();
        } catch (IOException e) {
            return "Error: " + e.getMessage();
        }

    }

}
